File: packages/js-dpp/lib/stateTransition/fees/operations/PreCalculatedOperation.js
const AbstractOperation = require('./AbstractOperation');

class PreCalculatedOperation extends AbstractOperation {
  /**
   * @param {number} storageCost
   * @param {number} processingCost
   */
  constructor(storageCost, processingCost) {
    super();

    this.storageCost = storageCost;
    this.processingCost = processingCost;
  }

  /**
   * Get CPU cost of the operation
   *
   * @returns {number}
   */
  getProcessingCost() {
    return this.processingCost;
  }

  /**
   * Get storage cost of the operation
   *
   * @returns {number}
   */
  getStorageCost() {
    return this.storageCost;
  }

  /**
   * Get operation type
   *
   * @returns {string}
   */
  getType() {
    return AbstractOperation.TYPES.PRE_CALCULATED;
  }
}

module.exports = PreCalculatedOperation;

File: packages/rs-dpp/src/state_transition/fees/operations/precalculated_operation.rs
use super::{OperationLike, OperationType};

pub const BASE_PROCESSING_COST: i64 = 20000;

#[derive(Default, Debug, Clone, Copy)]
pub struct PreCalculatedOperation {
    storage_cost: i64,
    processing_cost: i64,
}

impl PreCalculatedOperation {
    pub fn new(storage_cost: i64, processing_cost: i64) -> Self {
        Self {
            storage_cost,
            processing_cost,
        }
    }
}

impl OperationLike for PreCalculatedOperation {
    fn get_processing_cost(&self) -> i64 {
        self.processing_cost
    }

    fn get_storage_cost(&self) -> i64 {
        self.storage_cost
    }

    fn get_type(&self) -> OperationType {
        OperationType::PreCalculated
    }
}
